ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -315,7 +319,15 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
         Transmission transmission = getTransmission(key);
         try {
           if (key.isConnectable()) {
-            handleConnect(key, transmission);
+            transmission.finishConnect();
+            if (transmission.ready()) {
+              connected.add(transmission.getConnectionId());
+              metrics.selectorConnectionCreated.inc();
+            } else {
+              pendingSslTrans.put(transmission.getConnectionId(),
```

Comment:
this assumes that any transmission that is connected but not ready is an `SSLTransmission`. While that may be true I am not so sure the `Selector` can make that assumption. I don't think it needs to make that assumption, really. The `TransmissionPendingHandshake` class could just be `UnreadyConnections` instead - and you could avoid referring to "handshake" everywhere.